src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -77,12 +80,20 @@ public BusinessesService(List<Listing> allBusinesses) {
             .apiKey(KEY)
             .build();
     TextSearchRequest request = PlacesApi.textSearchQuery(context, product);
-    
+    int radius = INITIAL_RADIUS;
+
     try {
       PlacesSearchResponse response = request.location(latLng)
-              .radius(TEXT_SEARCH_RADIUS)
+              .radius(radius)
               .await();
-      
+
+      while (response.results.length < MAX_NUMBER_OF_RESULTS_PER_REQUEST && 
+             radius < MAX_ALLOWED_TEXT_SEARCH_RADIUS) {
+        // Increase radius if there are less than 20 results 
+        radius = INITIAL_RADIUS * RADIUS_MULTIPLIER;
```

Comment:
You still want radius *= Radius_MULTIPLIER.

Reply:
Yep that makes sense